src/auto/injector.js

```diff
@@ -819,7 +818,8 @@ function createInjector(modulesToLoad, strictDi) {
         try {
           path.unshift(serviceName);
           cache[serviceName] = INSTANTIATING;
-          return cache[serviceName] = factory(serviceName, caller);
+          cache[serviceName] = factory(serviceName, caller);
```

Comment:
~100 lines up (in `provider`) this was fixed by adding `(....)` around the expression instead. Why solve this one differently?